src/story_protocol_python_sdk/resources/Group.py

```diff
@@ -0,0 +1,652 @@
+            # Validate licensing config
+            licensing_config = item.get('licensing_config', {})
+            
+            try:
+                self.license_terms_util.validate_licensing_config(licensing_config)
+            except Exception as e:
+                raise ValueError(f"Licensing config validation failed: {str(e)}")
+            
+            # Convert to camelCase for contract interaction
+            camelcase_config = {
+                'isSet': licensing_config.get('is_set', True),
+                'mintingFee': licensing_config.get('minting_fee', 0),
+                'hookData': licensing_config.get('hook_data', ZERO_ADDRESS),
+                'licensingHook': licensing_config.get('licensing_hook', ZERO_ADDRESS),
+                'commercialRevShare': licensing_config.get('commercial_rev_share', 0),
+                'disabled': licensing_config.get('disabled', False),
+                'expectMinimumGroupRewardShare': licensing_config.get('expect_minimum_group_reward_share', 0),
+                'expectGroupRewardPool': licensing_config.get('expect_group_reward_pool', ZERO_ADDRESS)
+            }
+            
+            processed_item = {
+                'licenseTemplate': license_template,
+                'licenseTermsId': item['license_terms_id'],
+                'licensingConfig': camelcase_config
+            }
+            
+            result.append(processed_item)
+        
+        return result
+
+    def _get_ip_metadata(self, ip_metadata: dict = None) -> dict:
+        """
+        Process IP metadata into the format expected by the contracts.
+
+        :param ip_metadata dict: [Optional] The metadata for the IP.
+        :return dict: Processed IP metadata.
+        """
+        metadata = {
+            'ipMetadataURI': "",
+            'ipMetadataHash': ZERO_HASH,
+            'nftMetadataURI': "",
+            'nftMetadataHash': ZERO_HASH,
+        }
+        
+        if ip_metadata:
+            metadata.update({
+                'ipMetadataURI': ip_metadata.get('ip_metadata_uri', ""),
+                'ipMetadataHash': ip_metadata.get('ip_metadata_hash', ZERO_HASH),
+                'nftMetadataURI': ip_metadata.get('nft_metadata_uri', ""),
+                'nftMetadataHash': ip_metadata.get('nft_metadata_hash', ZERO_HASH),
+            })
+        
+        return metadata
+
+    def _parse_tx_ip_group_registered_event(self, tx_receipt: dict) -> str:
+        """
+        Parse the IPGroupRegistered event from a transaction receipt.
+
+        :param tx_receipt dict: The transaction receipt.
+        :return str: The group ID.
+        """
+        event_signature = self.web3.keccak(text="IPGroupRegistered(address,address)").hex()
+        
+        for log in tx_receipt['logs']:
+            if log['topics'][0].hex() == event_signature:
+                group_id = '0x' + log['topics'][1].hex()[24:]
+                return self.web3.to_checksum_address(group_id)
+        
+        return None
```

Comment:
This function returns `None` but the documentation says `:return str: The group ID.`